### PR TITLE
fix(dashboard): Make sure fallback locale `en` also recognized by weather_status for F vs C usage

### LINKED ISSUE #23486: Unit for temperature in dashboard weather widget is always Celsius
**Is your feature request related to a problem? Please describe.**
The weather widget on the dashboard shows the temperature in Celsius, regardless of the locale. In the US, for example, I would expect it to show degrees Fahrenheit by default.

**Describe the solution you'd like**
Default to degrees Fahrenheit when the locale is `en-US`, or allow user to select their desired units.

### PR DESCRIPTION
**Pull request by joshtrichards**
# BROADER ISSUE DISCOVERED. On-hold until #49988 is reviewed
 

* Resolves: #23486 

## Summary

This should fix #23486. We use `en` as the fallback locale in numerous places. Since the underlying cause for this bug only occurs when the fallback is being relied upon seems to makes sense to include it here. Alternative would be to change the fallback locale to something like `en_US` across our code base, but that seems both overly drastic and unnecessary.

Refs: https://github.com/nextcloud/server/issues/23486#issuecomment-1629239056

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by joshtrichards** [comment]
The more I think about this, this is not the correct approach and something else is going on here.

We shouldn't even be hitting the fallback: we pull from the user session (ACCEPT_LANGUAGE) if we haven't saved a locale in their personal settings yet.

https://github.com/nextcloud-libraries/nextcloud-l10n/blob/985fe306ac6e859a2fe8c18afaa253845ae2624a/lib/locale.ts#L9

https://github.com/nextcloud/server/blob/d8d708e0a8b0bf12590406a4736e46e5d3373d24/lib/private/TemplateLayout.php#L190

https://github.com/nextcloud/server/blob/d8d708e0a8b0bf12590406a4736e46e5d3373d24/lib/private/L10N/Factory.php#L240

We seem to be losing the more specific `locale` or providing the wrong one to the front end because `data-locale` is ending up as `en` even when `en-us` is being sent by the browser.

```
Accept-language: en-US,en;q=0.5
```

Results in our template returning:

```
<html class="ng-csp" data-placeholder-focus="false" lang="en" data-locale="en" translate="no" >
```

And we appear to pick the right locale on the Personal settings page, which is why the problem goes away when selected and saved.

**Comment by joshtrichards** [comment]
As I suspected the problem is broader.

We're not showing the actual (active) `Locale` in the Personal settings page (unless it's actually been changed and re-saved).

## TLDR

The inconsistency is we call `findLocal()` two different ways, which produces different fallback return values and, creating further confusion, this leads to us showing the current (expected) Locale under Personal settings even though it really isn't the one it says.

Assuming we're comfortable with how locale handling is working in general, the fix just needs to happen in the Personal settings page.

### Details

For the same user (when they have not yet saved a locale in their preferences):

- `findLocale()` returns `en` as the locale in our templates (i.e. 99% of interactions)
- `findLocale()` returns `en_US` as the locale in the Personal settings page (which isn't what's really being used)

Cause:

In our template we call it as `findLocal(lang)` which means if a user doesn't have a locale saved yet we hit this fallback code:

https://github.com/nextcloud/server/blob/d8d708e0a8b0bf12590406a4736e46e5d3373d24/lib/private/L10N/Factory.php#L267-L269

This returns the language (`en` in my test case) as the fallback locale under in all circumstances. This is why `data-locale` is set to `en`.

But when the user loads up their Personal settings we populate the active locale without providing the language clue to `findLocale()`:

https://github.com/nextcloud/server/blob/d8d708e0a8b0bf12590406a4736e46e5d3373d24/apps/settings/lib/Settings/Personal/PersonalInfo.php#L268

...So we hit a different fallback:

https://github.com/nextcloud/server/blob/d8d708e0a8b0bf12590406a4736e46e5d3373d24/lib/private/L10N/Factory.php#L272-L274

And thus return a different initial locale for the same user. 

This finally explains why the Personal settings page always shows me the right thing, but behavior elsewhere in Nextcloud doesn't match it.

### Fixing

So I think we have two paths to consider here:

1. When the user doesn't have a saved locale yet, always just use their base language (i.e. `en` in my test case). This is effectively what we're doing throughout Nextcloud at the moment *other than on the Personal settings page*. If that is considered reasonable (presumably it is), then I propose we fix the call to `findLocale()` on the Personal settings page to be consistent (otherwise it is effectively displaying incorrect information to the user until/unless they save their locale preferences) [Note: There's a chance this also explains some other locale related bugs; still looking through those)

2. We remove the language value provided to `findLocal()` and make it consistent with the Personal settings page handling. In that case we'd be effectively changing *everyone* that doesn't have a saved locale yet to `en_US` <-- don't like this

**Comment by joshtrichards** [comment]
Superseded by #49988
